compiler/rustc_session/src/config.rs

```diff
@@ -963,6 +965,12 @@ fn default_configuration(sess: &Session) -> CrateConfig {
             }
         }
     }
+    if has_atomic {
+        ret.insert((sym::target_has_atomic_load_store, None));
+        if atomic_cas {
+            ret.insert((sym::target_has_atomic, None));
```

Comment:
This will likely need to be emitted only on nightly to avoid immediately stabilizing it.

Reply:
But then we will need to keep the `#[cfg(target_has_atomic]` annotations only for nightly as well, without stabilizing them? How do we do that?

And do you mean using

```rust
if sess.is_nightly_build() and has_atomic {
```

here?

Reply:
> Libcore can use nightly features even on stable rustc.

Oh, I didn't know about that. I thought that adding the `target_has_atomic` symbol only for nightly would break the core build with stage0. But it seems like it works.

So I applied only that one line change. PTAL

Reply:
Thanks!